### PR TITLE
ci(schema): auto commit schema changes

### PR DESCRIPTION
**Pull request by aali309**
# Welcome to Cryostat! 👋
## Before contributing, make sure you have:
* [x] Read the [contributing guidelines](https://github.com/cryostatio/cryostat/blob/main/CONTRIBUTING.md)
* [x] Linked a relevant issue which this PR resolves
* [x] Linked any other relevant issues, PR's, or documentation, if any
* [x] Resolved all conflicts, if any
* [x] Rebased your branch PR on top of the latest upstream `main` branch
* [x] Attached at least one of the following labels to the PR: `[chore, ci, docs, feat, fix, test]`
* [x] [Signed all commits using a GPG signature](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification#gpg-commit-signature-verification)

**To recreate commits with GPG signature** `git fetch upstream && git rebase --force --gpg-sign upstream/main`
_______________________________________________

Fixes: #377 

## Description of the change:
*This change allows an environment variable to be configured so that...*

## Motivation for the change:
*This change is helpful because users may want to...*

## How to manually test:
1. *Run CRYOSTAT_IMAGE=quay.io... bash smoketest.bash...*
2. *...*

### CODE REVIEW

**Comment by aali309** [comment]
sample test run:
[https://github.com/aali309/cryostat/pull/34/commits/fdcc905087a05917a8139c115a4b0929936bfca2](https://github.com/aali309/cryostat/pull/34/commits/fdcc905087a05917a8139c115a4b0929936bfca2)

Actions link: Added `ls` and `cat` commands for verification here.
[https://github.com/aali309/cryostat/actions/runs/10050986434/job/27779902429](https://github.com/aali309/cryostat/actions/runs/10050986434/job/27779902429)

File: .github/workflows/pr-ci.yaml [hunk 1]
```diff
@@ -295,3 +295,56 @@ jobs:
...
+    if: needs.update-schemas.outputs.OPENAPI_STATUS != '0' || needs.update-schemas.outputs.GRAPHQL_STATUS != '0'
+    steps:
+      - name: Check out PR branch
+        uses: actions/checkout@v3
+        with:
+          repository: ${{ needs.checkout-branch.outputs.PR_repo }}
+          ref: ${{ needs.checkout-branch.outputs.PR_head_ref }}
+          submodules: true
+          fetch-depth: 0
+      - name: Set up Bot user
```
Comment (andrewazores) [inline .github/workflows/pr-ci.yaml:311]: Better to do it like this, I think:

https://github.com/cryostatio/cryostat-web/blob/7c0824c6a19515823d901c3a894441b9f4860fd9/.github/workflows/submodule.yaml#L32

this way the existing signing key for a CI bot account is used and these commits will be verified.

File: .github/workflows/pr-ci.yaml [hunk 2]
```diff
@@ -295,3 +295,56 @@ jobs:
...
+        run: |
+          patch -p1 < diffs/openapi.diff
+      - name: Apply GraphQL diff
+        if: needs.update-schemas.outputs.GRAPHQL_STATUS != '0'
+        run: |
+          patch -p1 < diffs/graphql.diff
+      - name: Commit and push schema changes
+        run: |
+          git add schema/openapi.yaml schema/schema.graphql
+          git commit -m "CI committing detected Schema changes"
```
Comment (andrewazores) [inline .github/workflows/pr-ci.yaml:336]: Maybe something like `chore(schema): automatic update`?

**Comment by aali309** [comment]
New test run
[https://github.com/aali309/cryostat/actions/runs/10061949995/job/27813420013](https://github.com/aali309/cryostat/actions/runs/10061949995/job/27813420013)

image

**Comment by andrewazores** [comment]
Strange that something is preventing Quarkus from starting up now...

**Comment by aali309** [comment]
> Strange that something is preventing Quarkus from starting up now...

Yes, everything looks good. Running the tests Locally too!

**Comment by andrewazores** [comment]
Maybe the container image that was built is somehow faulty... ?

**Review by andrewazores — APPROVED** [review]

**Comment by andrewazores** [comment]
Let's see if anything changes when this gets picked up the CI job to build it on push to `main`.

**Comment by andrewazores** [comment]
https://github.com/cryostatio/cryostat/actions/runs/10063123609/job/27817201937

**Comment by andrewazores** [comment]
Still failing. Now that I'm looking more closely however, I see this in the logs:

```
2024-07-23 17:03:50,684 INFO  [tc.doc.io/localstack/localstack:latest] (pool-6-thread-1) Creating container for image: docker.io/localstack/localstack:latest
2024-07-23 17:03:50,724 INFO  [tc.doc.io/localstack/localstack:latest] (pool-6-thread-1) Container docker.io/localstack/localstack:latest is starting: 57c2cba5b90b92603dafa09553801209e2fc0fb5f3d905bfa44a1fba00e5e9d1
2024-07-23 17:04:51,819 ERROR [tc.doc.io/localstack/localstack:latest] (pool-6-thread-1) Could not start container: org.testcontainers.containers.ContainerLaunchException: Timed out waiting for container to become healthy
	at org.testcontainers.containers.wait.strategy.DockerHealthcheckWaitStrategy.waitUntilReady(DockerHealthcheckWaitStrategy.java:25)
	at org.testcontainers.containers.wait.strategy.AbstractWaitStrategy.waitUntilReady(AbstractWaitStrategy.java:52)
	at org.testcontainers.containers.GenericContainer.waitUntilContainerStarted(GenericContainer.java:964)
	at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:490)
	at org.testcontainers.containers.GenericContainer.lambda$doStart$0(GenericContainer.java:344)
	at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:81)
	at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:334)
	at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:322)
	at io.cryostat.resources.LocalStackResource.start(LocalStackResource.java:54)
	at io.quarkus.test.common.TestResourceManager$TestResourceEntryRunnable.run(TestResourceManager.java:492)
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1804)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)
2024-07-23 17:04:51,875 ERROR [tc.doc.io/localstack/localstack:latest] (pool-6-thread-1) Log output from the failed container:
LocalStack version: 3.5.1.dev
LocalStack build date: 2024-07-23
LocalStack build git hash: a0a1ba090
Ready.
```

Looks like testcontainers and the latest localstack version aren't playing nice?

**Comment by andrewazores** [comment]
The `localstack:latest` image did get updated 3 hours ago, which lines up with our timeframe.

image

https://hub.docker.com/r/localstack/localstack/tags

**Pull request merged**